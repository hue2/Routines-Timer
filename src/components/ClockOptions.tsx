import * as React from 'react';
import '../styles/App.css';
import '../styles/Options.css';

type ClockOptionsState = {
    show: boolean,
    closeModal?: () => void;
    maxLength: number,
}

type ClockOptionsProps = {
    navOpen: boolean,
    navClose: () => void,
    setHour: (event: any) => void,
    setMinute: (event: any) => void,
    setSeconds: (event: any) => void,
    setMinuteBreak: (event: any) => void,
    setSecondsBreak: (event: any) => void,
    setRepeats: (event: any) => void,
}

const style = {
    open: {
        width: "20%"
    },
    close: {
        width: "0%"
    }
}


export default class ClockOptions extends React.Component<ClockOptionsProps, ClockOptionsState>  {
    constructor(props: ClockOptionsProps) {
        super(props);
        this.state = { show: false, maxLength: 3 }
    }

    openNav = () => {}
    setTimer = () => {}

    render() {
        return(
            <div id="myNav" className="overlay" style={this.props.navOpen ? style.open : style.close }>
            <a href="javascript:void(0)" className="closebtn" onClick={this.props.navClose}>&times;</a>
                <div className="overlay-content">
                    <div className="option-div">
                        <span className="option-label">Time: </span>
                        <input type="text" placeholder="hh" maxLength={this.state.maxLength} onBlur={this.props.setHour}/>
                        <input className="dot-separator" type="text" placeholder=":" disabled/>
                        <input type="text" placeholder="mm" maxLength={this.state.maxLength} onBlur={this.props.setMinute}/>
                        <input className="dot-separator" type="text" placeholder=":" disabled/>
                        <input type="text" placeholder="ss" maxLength={this.state.maxLength} onBlur={this.props.setSeconds}/>
                    </div>
                    <div className="option-div">
                        <span className="option-label">Breaks: </span>
                        <input type="text" placeholder="mm" maxLength={this.state.maxLength} onBlur={this.props.setMinuteBreak}/>
                        <input className="dot-separator" type="text" placeholder=":" disabled/>
                        <input type="text" placeholder="ss" maxLength={this.state.maxLength} onBlur={this.props.setSecondsBreak}/>
                    </div>

                    <div className="option-div">
                        <span className="option-label">Repeats: </span>
                        <input type="text" maxLength={this.state.maxLength} onBlur={this.props.setRepeats}/>
                    </div>
                 
                </div>
            </div>
        );
    }
}