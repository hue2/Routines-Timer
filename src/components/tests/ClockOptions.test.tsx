import * as React from 'react';
import {shallow} from 'enzyme';
import ClockOptions from '../ClockOptions';
import * as renderer from 'react-test-renderer';

describe('Clock Options Tests', () => {
    let props;

    beforeEach(() => {
        props = {
            time: {
                hour: 1, 
                second: 1, 
                minute: 1
            },
            breakTime: {
                minuteBreak: 1,
                secondBreak: 1
            },
            repeats: 0,
            navOpen: true,
            toggleNav: jest.fn(),
            handleChange: jest.fn(),
            handleTimeChange: jest.fn(),
        }
    });

    it('component matches snapshot', () => {
        const tree = renderer.create(
            <ClockOptions 
                {...props}
            />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should close nav bar is when clicked on the x button', () => {
        const wrapper = shallow(<ClockOptions 
            {...props}
        />);
        wrapper.find('.closebtn').simulate('click');
        expect(props.toggleNav).toHaveBeenCalled();
    });

    it('should call handleTimeChange on time input', () => {
        const wrapper = shallow(<ClockOptions 
            {...props}
        />);

        wrapper.find('.hour').simulate('change', { target: { value: 1 } });    
        expect(props.handleTimeChange).toHaveBeenCalled();
    });

    it('should call handleChange on break time input', () => {
        const wrapper = shallow(<ClockOptions 
            {...props}
        />);

        wrapper.find('.minuteBreak').simulate('change', { target: { value: 1 } });    
        expect(props.handleChange).toHaveBeenCalled();
    });

    it('should call handleChange on repeat input', () => {
        const wrapper = shallow(<ClockOptions 
            {...props}
        />);

        wrapper.find('.repeat').simulate('change', { target: { value: 1 } });    
        expect(props.handleChange).toHaveBeenCalled();
    });
})