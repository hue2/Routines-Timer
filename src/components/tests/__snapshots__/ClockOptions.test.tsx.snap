// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock Options Tests component matches snapshot 1`] = `
<div
  className="overlay open"
  id="myNav"
>
  <a
    className="closebtn"
    href="javascript:void(0)"
    onClick={[Function]}
  >
    Ã—
  </a>
  <div
    className="overlay-content"
  >
    <div
      className="option-div"
    >
      <div
        className="option-label"
      >
        Time: 
      </div>
      <div
        className="input-time"
      >
        <input
          className="hour"
          data-state="time"
          maxLength={3}
          min="1"
          onChange={[Function]}
          placeholder="hh"
          type="number"
          value={1}
        />
        <input
          className="dot-separator"
          disabled={true}
          placeholder=":"
          type="text"
        />
        <input
          className="minute"
          data-state="time"
          maxLength={3}
          min="1"
          onChange={[Function]}
          placeholder="mm"
          type="number"
          value={1}
        />
        <input
          className="dot-separator"
          disabled={true}
          placeholder=":"
          type="text"
        />
        <input
          className="second"
          data-state="time"
          maxLength={3}
          min="1"
          onChange={[Function]}
          placeholder="ss"
          type="number"
          value={1}
        />
      </div>
    </div>
    <div
      className="option-div"
    >
      <div
        className="option-label"
      >
        Breaks: 
      </div>
      <div
        className="input-time"
      >
        <input
          className="minuteBreak"
          data-state="timeBreak"
          maxLength={3}
          min="1"
          onChange={[Function]}
          placeholder="mm"
          type="number"
          value={1}
        />
        <input
          className="dot-separator"
          disabled={true}
          placeholder=":"
          type="text"
        />
        <input
          className="secondBreak"
          data-state="timeBreak"
          maxLength={3}
          min="1"
          onChange={[Function]}
          placeholder="ss"
          type="number"
          value={1}
        />
      </div>
    </div>
    <div
      className="option-div"
    >
      <div
        className="option-label"
      >
        Repeats: 
      </div>
      <div
        className="input-time"
      >
        <input
          className="repeat tempRepeat"
          data-state="repeats"
          maxLength={3}
          min="1"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
